File: java/samples/PeoplePickerSample.java
package io.fluentcards.samples;

import io.fluentcards.*;
import java.util.Map;

public class PeoplePickerSample {
    public static void run() {
        Map<String, Object> card = AdaptiveCardBuilder.create()
            .withVersion("1.6")
            .addTextBlock(tb -> tb
                .withText("People Picker")
                .withSize(TextSize.LARGE)
                .withWeight(TextWeight.BOLDER))
            .addInputChoiceSet(i -> i
                .withId("people-picker")
                .withLabel("Select users in the whole organization")
                .withIsMultiSelect(true)
                .withStyle(ChoiceInputStyle.FILTERED)
                .withChoicesData("graph.microsoft.com/users"))
            .addAction(a -> a
                .submit("Submit"))
            .build();

        System.out.println("=== People Picker Card ===");
        System.out.println(CardSerializer.toJson(card));
    }
}

File: node/samples/deno/people_picker_sample.ts
// Deno sample: People picker card using dynamic datasets
// 
// To run: deno run people_picker_sample.ts

import { AdaptiveCardBuilder } from 'jsr:@adaptivecards/fluent';

/** Creates a people picker card that searches users from Microsoft Graph. */
export function createPeoplePickerCard() {
  return AdaptiveCardBuilder.create()
    .withVersion('1.6')
    .addInputChoiceSet((i) =>
      i
        .withId('people-picker')
        .withLabel('Select users in the whole organization')
        .withIsMultiSelect()
        .withValue('user1,user2')
        .withChoicesData('graph.microsoft.com/users'),
    )
    .addAction((a) => a.submit('Submit'))
    .build();
}

// Run when invoked directly
if (import.meta.main) {
  console.log('=== People Picker Sample ===\n');
  console.log('People Picker Card:', JSON.stringify(createPeoplePickerCard(), null, 2));
}
